pettingzoo/test/api_test.py

```diff
@@ -333,10 +330,14 @@ def play_test(env, observation_0, num_cycles):
         prev_observe, reward, terminated, truncated, info = env.last()
         if terminated or truncated:
             action = None
-        elif isinstance(prev_observe, dict) and "action_mask" in prev_observe:
-            action = random.choice(np.flatnonzero(prev_observe["action_mask"]).tolist())
         else:
-            action = env.action_space(agent).sample()
+            if isinstance(prev_observe, dict) and "action_mask" in prev_observe:
+                mask = prev_observe["action_mask"]
+            elif isinstance(info, dict) and "action_mask" in info:
+                mask = info["action_mask"]
+            else:
+                mask = None
+            action = env.action_space(agent).sample(mask)
```

Comment:
Can we not have an if loop within the else?

Reply:
I removed the info thing because that wasn’t actually used by pettingzoo. I believe your shimmy openspiel compatibility wrapper has action mask in info by default which was what threw me off. Again as above, this stuff is all pretty poorly designed and should be made consistent imo.

Reply:
Sorry, I meant this more as a stylistic thing, line 333 is an `else:`, then within this `else:` is this if loop. Could we not flatten this?

Reply:
I'll have a look at this again in the near future, along with the comment above this one.But on first glance it looks like I should put action mask into the observation for openspiel on shimmy.